src/dag/put.js

```diff
@@ -15,29 +14,35 @@ module.exports = (send) => {
 
   return promisify((dagNode, options, callback) => {
     if (typeof options === 'function') {
-      return setImmediate(() => callback(new Error('no options were passed')))
+      callback = options
+    } else if (options.cid && (options.format || options.hash)) {
```

Comment:
Notice that `js-ipfs-api`'s implementation expects `options.hash` instead of `options.hashAlg`. This seems to be inconsistent with the speck as per https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/DAG.md#javascript---ipfsdagputdagnode-options-callback

Changing this would be a breaking change. @alanshaw how do you feel about this?